code/_onclick/other_mobs.dm

```diff
@@ -53,9 +53,11 @@
 		A.attack_stump(src, params)
 
 	if(src.lying && !(isUnconscious() || stunned || paralysis) && check_crawl_ability() && isfloor(A) && isfloor(get_turf(src)) && proximity && !pulledby && !locked_to && !client.move_delayer.blocked())
-		var/crawldelay = movement_delay()*3
-		Move(A, get_dir(src,A), glide_size_override = crawldelay)
-		delayNextMove(crawldelay,additive=1)
+		var/delay = round(1 + base_movement_tally()/5) * 1 SECONDS
+		if (do_after(src, A, delay))
+			var/crawldelay = movement_delay()*3
+			Move(A, get_dir(src,A), glide_size_override = delay) // So that we're still smooth
+			delayNextMove(crawldelay,additive=1)
```

Comment:
Is there a reason to use `crawldelay` here now rather than the new `delay`?

Reply:
Not really, now that I think of it, given that `movement_delay()` and `base_movement_tally()` aren't so different.